### PR TITLE
Split to separate packages

### PR DESCRIPTION
**Pull request by roeepolegfiverr**

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -62,28 +62,28 @@ func Log(err error, level int) {
 }
 
 // RecoverAndLog catches an error, log it and recover. Return a gin.HandlerFunc
-func RecoverAndLog() gin.HandlerFunc {
+// func RecoverAndLog() gin.HandlerFunc {
```
Comment (nitsanfiverr) [inline logger/logger.go]: Why ? don't you want to try it ?

File: setup/setup.go [hunk 2]
```diff
@@ -1,38 +0,0 @@
-package shared
```
Comment (nitsanfiverr) [inline setup/setup.go:1]: Why you deleted this ?

File: statsd/statsd.go [hunk 3]
```diff
@@ -23,32 +23,28 @@ var (
...
 
-	}
-}
+// 	}
+// }
 
 // StatsDWrapper wraps the worker work to it can measure its times and other stats.
 // it will return a func so we can continue and wrap it with other wrappers, such as logger.
-func StatsDWrapper(o options, worker wrappedFn) wrappedFn {
+func StatsDWrapper(o logger.Options, worker logger.WrappedFn) logger.WrappedFn {
```
Comment (nitsanfiverr) [inline statsd/statsd.go]: maybe it's good to seperate it from logger ? it was easy on shared. here it has no meaning.

File: statsd/statsd.go [hunk 4]
```diff
@@ -68,11 +64,11 @@ func logWork(elapsed time.Duration, err error, o options) {
...
 	if val, ok := o["event"]; ok {
 		event := val.(string)
-		go sendMetrics(status, event, elapsed)
+		go SendMetrics(status, event, elapsed)
 	}
 }
 
-func sendMetrics(status string, event string, elapsed time.Duration) {
+func SendMetrics(status string, event string, elapsed time.Duration) {
 	incrementWorkCounters(status, event)
```
Comment (nitsanfiverr) [inline statsd/statsd.go:73]: maybe we should return the go inside the method

File: setup/setup.go (hunk 2 above)
Comment (roeepolegfiverr) [inline setup/setup.go:1]: not needed here, it will be part of the template

**Pull request closed without merging**